File: src/main/java/com/barberfind/api/shared/util/Normalizers.java
package com.barberfind.api.shared.util;

public final class Normalizers {
    private Normalizers() {}

    public static String normalizeEmail(String email) {
        return email == null ? null : email.trim().toLowerCase();
    }

    // mantém só dígitos; ex: "(55) 119999-0000" => "551199990000"
    public static String normalizePhone(String phone) {
        return phone == null ? null : phone.replaceAll("\\D", "");
    }
}

File: src/common/normalizers.ts
export function normalizeEmail(email: string | null | undefined): string | null {
  if (email == null) return null;
  return email.trim().toLowerCase();
}

/** Digits only, e.g. "(55) 11999-0000" -> "55119990000" */
export function normalizePhone(phone: string | null | undefined): string | null {
  if (phone == null) return null;
  return phone.replace(/\D/g, '');
}
